src/app/mesh/qgsmeshlayerproperties.cpp

```diff
@@ -0,0 +1,197 @@
+/***************************************************************************
+  qgsmeshlayerproperties.cpp
+  --------------------------
+    begin                : Jun 2018
+    copyright            : (C) 2018 by Peter Petrik
+    email                : zilolv at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <limits>
+#include <typeinfo>
+
+#include "qgisapp.h"
+#include "qgsapplication.h"
+#include "qgscoordinatetransform.h"
+#include "qgshelp.h"
+#include "qgslogger.h"
+#include "qgsmapcanvas.h"
+#include "qgsmaplayerstylemanager.h"
+#include "qgsmeshlayer.h"
+#include "qgsmeshlayerproperties.h"
+#include "qgsproject.h"
+#include "qgsprojectionselectiondialog.h"
+#include "qgsrenderermeshpropertieswidget.h"
+#include "qgssettings.h"
+
+#include <QFileDialog>
+
+QgsMeshLayerProperties::QgsMeshLayerProperties( QgsMapLayer *lyr, QgsMapCanvas *canvas, QWidget *parent, Qt::WindowFlags fl )
+  : QgsOptionsDialogBase( QStringLiteral( "MeshLayerProperties" ), parent, fl )
+  , mMeshLayer( qobject_cast<QgsMeshLayer *>( lyr ) )
```

Comment:
` if ( mMeshLayer )` is in the code because of pointer in the constuctor, not because of type cast to mesh layer. Do we want to replace it with reference?

Reply:
Until now probably everywhere in qgis code we use pointers to map layers, so i would keep that for consistency. Maybe just add Q_ASSERT( mMeshLayer ) to the constructor?